### PR TITLE
MINOR: fix code listings security.html

### PR DESCRIPTION
**Pull request by jlprat**
Fix examples under security.html so they use the right bash icon (`>`
instead of `$`) and also uses the right tool for showing code listings

Some of the diffs are caused also by the IDE aligning html code properly.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by jlprat** [comment]
CC. @cadonna as you reviewed the last doc changes on formatting and styling, you might want to review these as well. I split the changes in several PRs so it's more bearable to review.

File: docs/security.html [hunk 1]
```diff
@@ -428,12 +428,12 @@ <h3 class="anchor-heading"><a id="security_sasl" class="anchor-link"></a><a href
             <tt>zookeeper.sasl.client.username</tt> to the appropriate name
             (<i>e.g.</i>, <tt>-Dzookeeper.sasl.client.username=zk</tt>).</p>
 
-            <p>Brokers may also configure JAAS using the broker configuration property <code>sasl.jaas.config</code>.
-            The property name must be prefixed with the listener prefix including the SASL mechanism,
-            i.e. <code>listener.name.{listenerName}.{saslMechanism}.sasl.jaas.config</code>. Only one
-            login module may be specified in the config value. If multiple mechanisms are configured on a
-            listener, configs must be provided for each mechanism using the listener and mechanism prefix.
-            For example,
+                    <p>Brokers may also configure JAAS using the broker configuration property <code>sasl.jaas.config</code>.
```
Comment (showuon) [inline docs/security.html:431]: Line 402-408 has `ol` and `li` tag not get handled. Please help. Thanks.
https://github.com/apache/kafka/pull/10770/files#diff-3485b37e32662f4925ee0374c4f6eb1d4dfa589bbb1c148b2a70e92b4acbe8bdR402-R408

File: docs/security.html [hunk 2]
```diff
@@ -1835,28 +1834,28 @@ <h5 class="anchor-heading"><a id="operations_resources_and_protocols" class="anc
             <td></td>
             <td></td>
             <td>Creating delegation tokens has special rules, for this please see the
-                <a id="security_delegation_token" href="#security_delegation_token">Authentication using Delegation Tokens</a> section.</td>
+                <a id="security_delegation_token_1" href="#security_delegation_token">Authentication using Delegation Tokens</a> section.</td>
```
Comment (showuon) [inline docs/security.html:1837]: Nice fix!

**Review by showuon — COMMENTED** [review]
@jlprat , thanks for the PR. I had a quick look, and left some comments.

File: docs/security.html (hunk 1 above)
Comment (jlprat) [reply docs/security.html:431]: That `li` tag is currently closed at line 508. It's a 3 level nested ordered lists. Between line 402 and 508 we have the whole sublists as part of that `li` element. That's the reason the closing `li` element seems to be missing, it's just really down the page.

image

File: docs/security.html [hunk 3]
```diff
@@ -47,7 +47,7 @@ <h3 class="anchor-heading"><a id="security_ssl" class="anchor-link"></a><a href=
             The tool supports two different keystore formats, the Java specific jks format which has been deprecated by now, as well as PKCS12.
             PKCS12 is the default format as of Java version 9, to ensure this format is being used regardless of the Java version in use all following
             commands explicitly specify the PKCS12 format.
-            <pre class="line-numbers"><code class="language-bash">keytool -keystore {keystorefile} -alias localhost -validity {validity} -genkey -keyalg RSA -storetype pkcs12</code></pre>
+            <pre class="line-numbers"><code class="language-bash">&gt; keytool -keystore {keystorefile} -alias localhost -validity {validity} -genkey -keyalg RSA -storetype pkcs12</code></pre>
```
Comment (cadonna) [inline docs/security.html:50]: I am wondering if it is a good idea to add `>` in front of the commands. It is something additional users need to remove when they copy & paste the commands.

File: docs/security.html [hunk 4]
```diff
@@ -208,25 +208,25 @@ <h5>Host Name Verification</h5>
             Then create a database and serial number file, these will be used to keep track of which certificates were signed with this CA. Both of
             these are simply text files that reside in the same directory as your CA keys.
 
-            <pre class="line-numbers"><code class="language-bash">echo 01 > serial.txt
+            <pre class="line-numbers"><code class="language-bash">&gt; echo 01 &gt; serial.txt
 touch index.txt</code></pre>
```
Comment (cadonna) [inline docs/security.html]: If we decide to keep the `>`, then it is missing here.

File: docs/security.html [hunk 5]
```diff
@@ -237,10 +237,10 @@ <h5>Host Name Verification</h5>
         </li>
         <li><h4 class="anchor-heading"><a id="security_ssl_signing" class="anchor-link"></a><a href="#security_ssl_signing">Signing the certificate</a></h4>
             Then sign it with the CA:
-            <pre class="line-numbers"><code class="language-bash">openssl ca -config openssl-ca.cnf -policy signing_policy -extensions signing_req -out {server certificate} -infiles {certificate signing request}</code></pre>
+            <pre class="line-numbers"><code class="language-bash">&gt; openssl ca -config openssl-ca.cnf -policy signing_policy -extensions signing_req -out {server certificate} -infiles {certificate signing request}</code></pre>
 
             Finally, you need to import both the certificate of the CA and the signed certificate into the keystore:
-            <pre class="line-numbers"><code class="language-bash">keytool -keystore {keystore} -alias CARoot -import -file {CA certificate}
+            <pre class="line-numbers"><code class="language-bash">&gt; keytool -keystore {keystore} -alias CARoot -import -file {CA certificate}
 keytool -keystore {keystore} -alias localhost -import -file cert-signed</code></pre>
```
Comment (cadonna) [inline docs/security.html]: If we decide to keep the `>`, then it is missing here.

File: docs/security.html [hunk 6]
```diff
@@ -384,16 +384,16 @@ <h5>SSL key and certificates in PEM format</h5>
...
+                <li>ssl.cipher.suites (Optional). A cipher suite is a named combination of authentication, encryption, MAC and key exchange algorithm used to negotiate the security settings for a network connection using TLS or SSL network protocol.</li>
+                <li>ssl.enabled.protocols=TLSv1.2,TLSv1.1,TLSv1. It should list at least one of the protocols configured on the broker side</li>
+                <li>ssl.truststore.type=JKS</li>
+                <li>ssl.keystore.type=JKS</li>
+            </ol>
+            <br>
             Examples using console-producer and console-consumer:
-            <pre class="line-numbers"><code class="language-bash">kafka-console-producer.sh --bootstrap-server localhost:9093 --topic test --producer.config client-ssl.properties
+            <pre class="line-numbers"><code class="language-bash">&gt; kafka-console-producer.sh --bootstrap-server localhost:9093 --topic test --producer.config client-ssl.properties
 kafka-console-consumer.sh --bootstrap-server localhost:9093 --topic test --consumer.config client-ssl.properties</code></pre>
```
Comment (cadonna) [inline docs/security.html]: If we decide to keep the `>`, then it is missing here.

File: docs/security.html [hunk 7]
```diff
@@ -428,12 +428,12 @@ <h3 class="anchor-heading"><a id="security_sasl" class="anchor-link"></a><a href
...
-            i.e. <code>listener.name.{listenerName}.{saslMechanism}.sasl.jaas.config</code>. Only one
-            login module may be specified in the config value. If multiple mechanisms are configured on a
-            listener, configs must be provided for each mechanism using the listener and mechanism prefix.
-            For example,
+                    <p>Brokers may also configure JAAS using the broker configuration property <code>sasl.jaas.config</code>.
+                        The property name must be prefixed with the listener prefix including the SASL mechanism,
+                        i.e. <code>listener.name.{listenerName}.{saslMechanism}.sasl.jaas.config</code>. Only one
+                        login module may be specified in the config value. If multiple mechanisms are configured on a
+                        listener, configs must be provided for each mechanism using the listener and mechanism prefix.
+                        For example,
```
Comment (cadonna) [inline docs/security.html]: Why the indentation here? It should be on the same level as the previous paragraph, right?

File: docs/security.html [hunk 8]
```diff
@@ -443,134 +443,133 @@ <h3 class="anchor-heading"><a id="security_sasl" class="anchor-link"></a><a href
...
-                <li>Add a JAAS config file with a client login section named <tt>KafkaClient</tt>. Configure
-                    a login module in <tt>KafkaClient</tt> for the selected mechanism as described in the examples
-                    for setting up <a href="#security_sasl_kerberos_clientconfig">GSSAPI (Kerberos)</a>,
-                    <a href="#security_sasl_plain_clientconfig">PLAIN</a>,
-                    <a href="#security_sasl_scram_clientconfig">SCRAM</a> or
-                    <a href="#security_sasl_oauthbearer_clientconfig">OAUTHBEARER</a>.
-                    For example, <a href="#security_sasl_gssapi_clientconfig">GSSAPI</a>
-                    credentials may be configured as:
-                    <pre class="line-numbers"><code class="language-text">KafkaClient {
+                    If JAAS configuration is defined at different levels, the order of precedence used is:
```
Comment (cadonna) [inline docs/security.html:446]: Also this indentation do not seem to align to the previous section header `JAAS configuration for Kafka brokers`.

File: docs/security.html (hunk 3 above)
Comment (cadonna) [reply docs/security.html:50]: Now I saw that it has been already used further down. I fine with keeping it to distingiush better between configs and commands. However, I found some places where `>` is still missing (see further down).

**Review by cadonna — COMMENTED** [review]
@jlprat Thank you!

Here my feedback!

The added indentations sometimes do not match the previous section at the same level.

File: docs/security.html (hunk 3 above)
Comment (jlprat) [reply docs/security.html:50]: The examples present in the documentation, use more frequently the `>` character when showing commands. Sometimes a `$` is shown, and really occasionally nothing is put in there.
As `>` was the most common used one already in the codebase, I decided to unify on this character.

File: docs/security.html (hunk 7 above)
Comment (jlprat) [reply docs/security.html]: Actually, it seems the above paragraphs should be indented to the right, there are several nested `<ol>` and `<li>` and `<p>` that didn't follow indentation.
I'm fixing these now.

File: docs/security.html (hunk 8 above)
Comment (jlprat) [reply docs/security.html:446]: Same as previous, you are right, it needs to be aligned with the contents of the header `JAAS configuration for Kafka brokers`. However, the problem here was that the previous content was misaligned.

**Comment by jlprat** [comment]
Hi @cadonna thanks a lot for the feedback. I fixed the missing `>` characters. About the misalignment, you were right, however it was not from the lines you mention, but the previous ones.

File: docs/security.html [hunk 9]
```diff
@@ -76,7 +76,7 @@ <h5>Host Name Verification</h5>
             Server host name verification may be disabled by setting <code>ssl.endpoint.identification.algorithm</code> to an empty string.<br>
             For dynamically configured broker listeners, hostname verification may be disabled using <code>kafka-configs.sh</code>:<br>
 
-            <pre class="line-numbers"><code class="language-text">bin/kafka-configs.sh --bootstrap-server localhost:9093 --entity-type brokers --entity-name 0 --alter --add-config "listener.name.internal.ssl.endpoint.identification.algorithm="</code></pre>
+            <pre class="line-numbers"><code class="language-text">&gt; bin/kafka-configs.sh --bootstrap-server localhost:9093 --entity-type brokers --entity-name 0 --alter --add-config "listener.name.internal.ssl.endpoint.identification.algorithm="</code></pre>
```
Comment (cadonna) [inline docs/security.html]: I think this should be `class="language-bash"` instead of `class="language-text"`.

File: docs/security.html [hunk 10]
```diff
@@ -99,7 +99,7 @@ <h5>Host Name Verification</h5>
 
 
             To add a SAN field append the following argument <code> -ext SAN=DNS:{FQDN},IP:{IPADDRESS} </code> to the keytool command:
```
Comment (cadonna) [inline docs/security.html]:
```suggestion
            To add a SAN field append the following argument <code> -ext SAN=DNS:{FQDN},IP:{IPADDRESS}</code> to the keytool command:
```

File: docs/security.html [hunk 11]
```diff
@@ -384,56 +384,56 @@ <h5>SSL key and certificates in PEM format</h5>
...
+                    brokers. If you want to use a section name other than Client, set the
+                    system property <tt>zookeeper.sasl.clientconfig</tt> to the appropriate
+                    name (<i>e.g.</i>, <tt>-Dzookeeper.sasl.clientconfig=ZkClient</tt>).</p>
+
+                    <p>ZooKeeper uses "zookeeper" as the service name by default. If you
+                    want to change this, set the system property
+                    <tt>zookeeper.sasl.client.username</tt> to the appropriate name
+                    (<i>e.g.</i>, <tt>-Dzookeeper.sasl.client.username=zk</tt>).</p>
+
+                    <p>Brokers may also configure JAAS using the broker configuration property <code>sasl.jaas.config</code>.
```
Comment (cadonna) [inline docs/security.html:431]: I could not find the `</p>` for this `<p>`.

File: docs/security.html (hunk 7 above)
Comment (cadonna) [reply docs/security.html]: I think we misunderstood each other. The line that starts with `The property name ...` should be aligned with `<p>Brokers may also ...`.  I am also fine, if you do not fix that. I think it is understandable and I see that the indentation is not really consistent in this file.

**Review by cadonna — COMMENTED** [review]
@jlprat Thank you for the update!

Here my feedback!

Once you answered to that, we can merge this PR and live with some indentation inconsistencies.

**Comment by jlprat** [comment]
Thanks @cadonna All comments addressed!

File: docs/security.html (hunk 11 above)
Comment (jlprat) [reply docs/security.html:431]: It was missing, good spot

**Comment by cadonna** [comment]
The test failures are unrelated since this is a pure doc change.

**Comment by jlprat** [comment]
Thanks @cadonna !

**Pull request merged**